python/cuml/manifold/t_sne.pyx

```diff
@@ -408,13 +442,38 @@ class TSNE(Base):
                                                        training data in \
                                                        low-dimensional space.',
                                        'shape': '(n_samples, n_components)'})
-    def fit_transform(self, X, convert_dtype=True):
-        """
-        Fit X into an embedded space and return that transformed output.
-
-
+    def fit_transform(self, X, convert_dtype=True, knn_graph=None):
+        """Fit X into an embedded space and return that transformed output.
+
+        Parameters
+        -----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            X contains a sample per row.
+        convert_dtype : bool, optional (default = True)
+            When set to True, the fit method will automatically
+            convert the inputs to np.float32.
+         knn_graph : sparse array-like (device or host)
+            shape=(n_samples, n_samples)
+            A sparse array containing the k-nearest neighbors of X,
+            where the columns are the nearest neighbor indices
+            for each row and the values are their distances.
+            Users using the knn_graph parameter provide t-SNE
+            with their own run of the KNN algorithm. This allows the user
+            to pick a custom distance function (sometimes useful
+            on certain datasets) whereas t-SNE uses euclidean by default.
+            The custom distance function should match the metric used
+            to train t-SNE embeedings. Storing and reusing a knn_graph
+            will also provide a speedup to the t-SNE algorithm
+            when performing a grid search.
+            Acceptable formats: sparse SciPy ndarray, CuPy device ndarray,
+            CSR/COO preferred other formats will go through conversion to CSR
+
+        Returns
+        --------
+        X_new : array, shape (n_samples, n_components)
+                Embedding of the training data in low-dimensional space.
         """
-        self.fit(X, convert_dtype=convert_dtype)
+        self.fit(X, convert_dtype, knn_graph)
```

Comment:
I would keep the named parameters here in case the order of the arguments should ever be switched around (that can be really hard to debug)
```suggestion
        self.fit(X, convert_dtype=convert_dtype, knn_graph=knn_graph)
```